Src/PCompiler/CompilerCore/Backend/PVerifier/Uclid5CodeGenerator.cs

```diff
@@ -424,29 +438,26 @@ private List<CompiledFile> CompileToFile(string name, ProofCommand cmd)
             files.Add(_src);
             GenerateMain(null, null, null, cmd);
         }
-        else
+        foreach (var m in machines)
         {
-            foreach (var m in machines)
+            if (_ctx.Job.CheckOnly == null || _ctx.Job.CheckOnly.Contains(m.Name))
             {
-                if (_ctx.Job.CheckOnly == null || _ctx.Job.CheckOnly.Contains(m.Name))
+                foreach (var s in m.States)
                 {
-                    foreach (var s in m.States)
+                    if (_ctx.Job.CheckOnly == null || _ctx.Job.CheckOnly.Contains(s.Name))
                     {
-                        if (_ctx.Job.CheckOnly == null || _ctx.Job.CheckOnly.Contains(s.Name))
-                        {
-                            _src = GenerateCompiledFile(cmd, name, m, s, null);
-                            files.Add(_src);
-                            GenerateMain(m, s, null, cmd);
-                        }
+                        _src = GenerateCompiledFile(cmd, name, m, s, null);
+                        files.Add(_src);
+                        GenerateMain(m, s, null, cmd, cmd.Name == "default");
+                    }
 
-                        foreach (var e in events.Where(e => !e.IsNullEvent && !e.IsHaltEvent && s.HasHandler(e)))
+                    foreach (var e in events.Where(e => !e.IsNullEvent && !e.IsHaltEvent && s.HasHandler(e)))
+                    {
+                        if (_ctx.Job.CheckOnly == null || _ctx.Job.CheckOnly.Contains(e.Name))
```

Comment:
What is the semantic of `CheckOnly` here? Seems we are checking both machine/state and event names. This might not work if I want to check all event handlers for a single machine (suppose I pass only the name of the machine to `CheckOnly`).

Reply:
@Ang9876

Reply:
I did not change this part of code but only remove `else`. My understanding is that the part corresponding to `machine` generates a file for `entry`, and that corresponding to `events` generates files for event handlers. So I guess there is no way to specify a single machine and check all event handlers (without checking event handlers of other machines). But I suspect that this CheckOnly is never set here.

Reply:
Let's keep it as is for now. I may change it later! Thanks!